lib/dynamoid/criteria/chain.rb

```diff
@@ -70,25 +70,27 @@ def last
 
       # Destroys all the records matching the criteria.
       #
-      def destroy_all
+      def delete_all
         ids = []
+        ranges = []
 
         if key_present?
-          ranges = []
           Dynamoid.adapter.query(source.table_name, range_query).collect do |hash|
             ids << hash[source.hash_key.to_sym]
-            ranges << hash[source.range_key.to_sym]
+            ranges << hash[source.range_key.to_sym] if source.range_key
           end
 
-          Dynamoid.adapter.delete(source.table_name, ids,{:range_key => ranges})
+          Dynamoid.adapter.delete(source.table_name, ids, range_key: ranges.presence)
```

Comment:
Oh nice, learned something new today, didn't realize `.presence` was a thing :) actually kind of handy!